File: src/lib/validations/consultation.ts
import { z } from "zod";

export const consultationCreateSchema = z.object({
  patientId: z.string().min(1, "Patient is required"),
  appointmentId: z.string().min(1, "Appointment is required"),
  chiefComplaint: z.string().optional().nullable(),
  symptoms: z.string().optional().nullable(),
  clinicalFindings: z.string().optional().nullable(),
  diagnosis: z.string().optional().nullable(),
  advice: z.string().optional().nullable(),
  followUpDate: z.coerce.date().optional().nullable(),
  notes: z.string().optional().nullable(),
});

export type ConsultationCreateInput = z.infer<typeof consultationCreateSchema>;

File: backend/app/schemas/consultation.py
from pydantic import BaseModel
from datetime import date, datetime


class ConsultationCreate(BaseModel):
    patient_id: str
    appointment_id: str
    chief_complaint: str | None = None
    history_of_present_illness: str | None = None
    past_medical_history: str | None = None
    symptoms: str | None = None
    clinical_findings: str | None = None
    diagnosis: str | None = None
    advice: str | None = None
    follow_up_date: date | None = None
    notes: str | None = None


class ConsultationUpdate(BaseModel):
    chief_complaint: str | None = None
    history_of_present_illness: str | None = None
    past_medical_history: str | None = None
    symptoms: str | None = None
    clinical_findings: str | None = None
    diagnosis: str | None = None
    advice: str | None = None
    follow_up_date: date | None = None
    notes: str | None = None


class ConsultationResponse(BaseModel):
    id: str
    patient_id: str
    appointment_id: str
    chief_complaint: str | None = None
    history_of_present_illness: str | None = None
    past_medical_history: str | None = None
    symptoms: str | None = None
    clinical_findings: str | None = None
    diagnosis: str | None = None
    advice: str | None = None
    follow_up_date: date | None = None
    notes: str | None = None
    is_signed: bool
    doctor_id: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
